telemetry/request.go

```diff
@@ -13,7 +13,7 @@ const (
 )
 
 type splittablePayloadEntry interface {
-	PayloadEntry
+	MapEntry
 	split() []splittablePayloadEntry
```

Comment:
I don't think this splitting logic works because `SpanBatch` doesn't implement this interface.